pkg/netpol/diff/diff_formatter.go

```diff
@@ -215,3 +215,130 @@ func (cs *diffFormatCSV) writeDiffOutput(connsDiff ConnectivityDiff) (string, er
+	// added
+	nPeers, newEdges, newIngressEdges := getEdgesAndPeersLinesByCategory(connsDiff.AddedConnections(), peersVisited)
+	peersLines = append(peersLines, nPeers...)
+	edgeLines = append(edgeLines, newEdges...)
+	ingressAnalyzerEdges = append(ingressAnalyzerEdges, newIngressEdges...)
+	// removed
+	lPeers, lostEdges, lostIngressEdges := getEdgesAndPeersLinesByCategory(connsDiff.RemovedConnections(), peersVisited)
+	peersLines = append(peersLines, lPeers...)
+	edgeLines = append(edgeLines, lostEdges...)
+	ingressAnalyzerEdges = append(ingressAnalyzerEdges, lostIngressEdges...)
+
+	// sort lines
+	sort.Strings(peersLines)
+	sort.Strings(edgeLines)
+	sort.Strings(ingressAnalyzerEdges)
+
+	// write graph
+	allLines := []string{dotHeader}
+	allLines = append(allLines, peersLines...)
+	allLines = append(allLines, edgeLines...)
+	allLines = append(allLines, ingressAnalyzerEdges...)
+	allLines = append(allLines, dotClosing)
+	return strings.Join(allLines, newLine), nil
+}
+
+func getEdgesAndPeersLinesByCategory(connsPairs []*ConnsPair, peersSet map[string]bool) (pLines, cEdges, iEdges []string) {
+	peersLines := make([]string, 0)
+	connsEdges := make([]string, 0)
+	ingressEdges := make([]string, 0)
+	for _, connsPair := range connsPairs {
+		src, dst, isIngress := getConnPeersStrings(connsPair)
+		// add peers lines (which still not in the set)
+		if !peersSet[src] {
+			peersSet[src] = true
+			peersLines = append(peersLines, addPeerLine(src, connsPair.diffType, connsPair.newOrLostSrc))
+		}
+		if !peersSet[dst] {
+			peersSet[dst] = true
+			peersLines = append(peersLines, addPeerLine(dst, connsPair.diffType, connsPair.newOrLostDst))
+		}
+		// add connections lines
+		if isIngress {
+			ingressEdges = append(ingressEdges, addEdgesLines(connsPair))
+		} else {
+			connsEdges = append(connsEdges, addEdgesLines(connsPair))
+		}
+	}
+	return peersLines, connsEdges, ingressEdges
+}
+
+const (
+	newPeerColor        = "green3"
+	removedPeerColor    = "red"
+	persistentPeerColor = "blue"
+)
+
+func addPeerLine(peerName, diffType string, isNewOrLost bool) string {
+	peerColor := persistentPeerColor
+	if isNewOrLost {
+		switch diffType {
+		case addedType:
+			peerColor = newPeerColor
+		case removedType:
+			peerColor = removedPeerColor
+		default: // will not get here
+			break
+		}
+	}
+	return fmt.Sprintf("\t%q [label=%q color=%q fontcolor=%q]", peerName, peerName, peerColor, peerColor)
+}
+
+const (
+	nonChangedConnColor = "grey"
+	changedConnColor    = "gold2"
+	removedConnColor    = "red2"
+	addedConnColor      = "green"
+)
+
+func addEdgesLines(connsPair *ConnsPair) string {
+	src, dst, _ := getConnPeersStrings(connsPair)
+	firstConn, secondConn := getDirsConnsStrings(connsPair)
+	switch connsPair.diffType {
+	case nonChangedType:
+		return getEdgeLine(src, dst, firstConn, nonChangedConnColor)
+	case changedType:
+		return getEdgeLine(src, dst, firstConn+"->"+secondConn, changedConnColor)
```

Comment:
can we change the format of changed conn edge?
I think `->` might be confusing
one option: `new_conn (was: old_conn)`
do you have other ideas?

Reply:
let's replace `was` with `old` instead

Reply:
done